src/main/java/com/google/impactdashboard/server/api_utilities/LogRetriever.java

```diff
@@ -56,34 +32,59 @@ private LogRetriever(LoggingClient logger) {
 
   /**
    * Creates a {@code ListLogEntriesRequest} and retrieves all the relevant audit logs.
-   * @return A list of all the relevant audit log entries that are stored by the logging API
+   * @param projectId ID of the project that the audit logs will be retrieved for.
+   * @param timeFrom The earliest time to retrieve logs for.
+   * @param timeTo The latest time to retrieve logs for.
+   * @return A response that contains all the relevant audit log entries that are stored by the logging API
    */
-  public Collection<LogEntry> listAuditLogs(String projectId) {
-    String project_id = "projects/" + projectId ; // needs to be retrieved from resource manager
-    // May need tweaking once tested and
-    String filter = "resource.type = project AND severity = NOTICE AND " +
-        "protoPayload.methodName:SetIamPolicy";
-    //Test of ListLogEntriesRequest will be changed once logging is tested
-    ListLogEntriesRequest request = ListLogEntriesRequest.newBuilder().setFilter(filter)
-        .setOrderBy("timestamp desc").addResourceNames(project_id).build();
-    ListLogEntriesPagedResponse response = logger.listLogEntries(request);
-    return StreamSupport.stream(response.iterateAll().spliterator(), false)
-        .collect(Collectors.toList());
+  public ListLogEntriesPagedResponse listAuditLogsResponse(String projectId, String timeFrom, String timeTo, int pageSize) {
+    String project_id = "projects/" + projectId;
+    String filter = "resource.type = project AND severity = NOTICE";
```

Comment:
StringBuilder would be the better practice here.

Reply:
I changed the string concat here to use string builder